From db358c127ff26c280bef5d85e62e3a08153438d5 Mon Sep 17 00:00:00 2001
From: Thierry Bordaz <tbordaz@redhat.com>
Date: Mon, 27 Jan 2020 13:49:37 +0100
Subject: [PATCH] Ticket 50857 - Memory leak in ACI using IP subject

Bug Description:
	When a ACI is evaluated (LASIpEval) a context (cookie) is allocated.
	At the end of the connection, the context is freed
	via a callback (LASIpFlush).
	The context contains two LASIpTree_t tree (ipv4 and ipv6)
	In free callback, only ipv4 tree is freed

Fix Description:
	Free ipv6 tree in LASIpTree

https://pagure.io/389-ds-base/issue/50857

Reviewed by: Mark Reynolds

Platforms tested: F31

Flag Day: no

Doc impact: no
---
 lib/libaccess/lasip.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/libaccess/lasip.cpp b/lib/libaccess/lasip.cpp
index 30c546df7..cdb88eec5 100644
--- a/lib/libaccess/lasip.cpp
+++ b/lib/libaccess/lasip.cpp
@@ -436,6 +436,7 @@ LASIpFlush(void **las_cookie)
         return;
 
     LASIpTreeDealloc(((LASIpContext_t *)*las_cookie)->treetop);
+    LASIpTreeDealloc(((LASIpContext_t *)*las_cookie)->treetop_ipv6);
     PERM_FREE(*las_cookie);
     *las_cookie = NULL;
     return;
-- 
2.24.1

